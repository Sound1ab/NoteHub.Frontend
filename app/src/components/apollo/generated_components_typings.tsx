/* tslint:disable */

type Maybe<T> = T | null
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string
  String: string
  Boolean: boolean
  Int: number
  Float: number
}

export type CreateNotebookInput = {
  userId: Scalars['ID']
  title: Scalars['String']
}

export type CreateNoteInput = {
  notebookId: Scalars['ID']
  title: Scalars['String']
  markdown: Scalars['String']
  excerpt: Scalars['String']
}

export type CreateUserInput = {
  firstName: Scalars['String']
  lastName: Scalars['String']
  email: Scalars['String']
}

export type Date = {
  dayOfWeek: Scalars['String']
  dayOfMonth: Scalars['Int']
  month: Scalars['String']
  dateLongForm: Scalars['String']
}

export type DeleteNotebookInput = {
  id?: Maybe<Scalars['ID']>
}

export type DeleteNoteInput = {
  id?: Maybe<Scalars['ID']>
}

export type DeleteUserInput = {
  id?: Maybe<Scalars['ID']>
}

export type ModelBooleanFilterInput = {
  ne?: Maybe<Scalars['Boolean']>
  eq?: Maybe<Scalars['Boolean']>
}

export type ModelFloatFilterInput = {
  ne?: Maybe<Scalars['Float']>
  eq?: Maybe<Scalars['Float']>
  le?: Maybe<Scalars['Float']>
  lt?: Maybe<Scalars['Float']>
  ge?: Maybe<Scalars['Float']>
  gt?: Maybe<Scalars['Float']>
  contains?: Maybe<Scalars['Float']>
  notContains?: Maybe<Scalars['Float']>
  between?: Maybe<Array<Maybe<Scalars['Float']>>>
}

export type ModelIdFilterInput = {
  ne?: Maybe<Scalars['ID']>
  eq?: Maybe<Scalars['ID']>
  le?: Maybe<Scalars['ID']>
  lt?: Maybe<Scalars['ID']>
  ge?: Maybe<Scalars['ID']>
  gt?: Maybe<Scalars['ID']>
  contains?: Maybe<Scalars['ID']>
  notContains?: Maybe<Scalars['ID']>
  between?: Maybe<Array<Maybe<Scalars['ID']>>>
  beginsWith?: Maybe<Scalars['ID']>
}

export type ModelIntFilterInput = {
  ne?: Maybe<Scalars['Int']>
  eq?: Maybe<Scalars['Int']>
  le?: Maybe<Scalars['Int']>
  lt?: Maybe<Scalars['Int']>
  ge?: Maybe<Scalars['Int']>
  gt?: Maybe<Scalars['Int']>
  contains?: Maybe<Scalars['Int']>
  notContains?: Maybe<Scalars['Int']>
  between?: Maybe<Array<Maybe<Scalars['Int']>>>
}

export type ModelNotebookConnection = {
  items?: Maybe<Array<Maybe<Notebook>>>
  nextOffset?: Maybe<Scalars['Int']>
}

export type ModelNotebookFilterInput = {
  id?: Maybe<ModelIdFilterInput>
  userId?: Maybe<ModelStringFilterInput>
}

export type ModelNoteConnection = {
  items?: Maybe<Array<Maybe<Note>>>
  nextToken?: Maybe<Scalars['String']>
}

export type ModelNoteFilterInput = {
  id?: Maybe<ModelIdFilterInput>
  description?: Maybe<ModelStringFilterInput>
  and?: Maybe<Array<Maybe<ModelNoteFilterInput>>>
  or?: Maybe<Array<Maybe<ModelNoteFilterInput>>>
  not?: Maybe<ModelNoteFilterInput>
}

export enum ModelSortDirection {
  Asc = 'ASC',
  Desc = 'DESC',
}

export type ModelStringFilterInput = {
  ne?: Maybe<Scalars['String']>
  eq?: Maybe<Scalars['String']>
  le?: Maybe<Scalars['String']>
  lt?: Maybe<Scalars['String']>
  ge?: Maybe<Scalars['String']>
  gt?: Maybe<Scalars['String']>
  contains?: Maybe<Scalars['String']>
  notContains?: Maybe<Scalars['String']>
  between?: Maybe<Array<Maybe<Scalars['String']>>>
  beginsWith?: Maybe<Scalars['String']>
}

export type ModelUserConnection = {
  items?: Maybe<Array<Maybe<User>>>
  nextToken?: Maybe<Scalars['String']>
}

export type ModelUserFilterInput = {
  id?: Maybe<ModelIdFilterInput>
  description?: Maybe<ModelStringFilterInput>
  and?: Maybe<Array<Maybe<ModelUserFilterInput>>>
  or?: Maybe<Array<Maybe<ModelUserFilterInput>>>
  not?: Maybe<ModelUserFilterInput>
}

export type Mutation = {
  createNote?: Maybe<Note>
  updateNote?: Maybe<Note>
  deleteNote?: Maybe<Note>
  createNotebook?: Maybe<Notebook>
  updateNotebook?: Maybe<Notebook>
  deleteNotebook?: Maybe<Notebook>
  createUser?: Maybe<User>
  updateUser?: Maybe<User>
  deleteUser?: Maybe<User>
}

export type MutationCreateNoteArgs = {
  input: CreateNoteInput
}

export type MutationUpdateNoteArgs = {
  input: UpdateNoteInput
}

export type MutationDeleteNoteArgs = {
  input: DeleteNoteInput
}

export type MutationCreateNotebookArgs = {
  input: CreateNotebookInput
}

export type MutationUpdateNotebookArgs = {
  input: UpdateNotebookInput
}

export type MutationDeleteNotebookArgs = {
  input: DeleteNotebookInput
}

export type MutationCreateUserArgs = {
  input: CreateUserInput
}

export type MutationUpdateUserArgs = {
  input: UpdateUserInput
}

export type MutationDeleteUserArgs = {
  input: DeleteUserInput
}

export type Note = {
  id: Scalars['ID']
  notebook: Notebook
  title: Scalars['String']
  markdown: Scalars['String']
  excerpt: Scalars['String']
  createdAt: Scalars['String']
  updatedAt: Scalars['String']
}

export type Notebook = {
  id: Scalars['ID']
  user: User
  title: Scalars['String']
  notes?: Maybe<Array<Maybe<Note>>>
  createdAt: Scalars['String']
  updatedAt: Scalars['String']
}

export type Query = {
  readNote?: Maybe<Note>
  listNotes?: Maybe<ModelNoteConnection>
  readNotebook?: Maybe<Notebook>
  listNotebooks?: Maybe<ModelNotebookConnection>
  readUser?: Maybe<User>
  listUsers?: Maybe<ModelUserConnection>
}

export type QueryReadNoteArgs = {
  id: Scalars['ID']
}

export type QueryListNotesArgs = {
  filter?: Maybe<ModelNoteFilterInput>
  limit?: Maybe<Scalars['Int']>
  nextToken?: Maybe<Scalars['String']>
}

export type QueryReadNotebookArgs = {
  id: Scalars['ID']
}

export type QueryListNotebooksArgs = {
  filter?: Maybe<ModelNotebookFilterInput>
  limit?: Maybe<Scalars['Int']>
  offset?: Maybe<Scalars['Int']>
}

export type QueryReadUserArgs = {
  id: Scalars['ID']
}

export type QueryListUsersArgs = {
  filter?: Maybe<ModelUserFilterInput>
  limit?: Maybe<Scalars['Int']>
  nextToken?: Maybe<Scalars['String']>
}

export type UpdateNotebookInput = {
  id: Scalars['ID']
  title?: Maybe<Scalars['String']>
}

export type UpdateNoteInput = {
  id: Scalars['ID']
  title?: Maybe<Scalars['String']>
  markdown?: Maybe<Scalars['String']>
  excerpt?: Maybe<Scalars['String']>
}

export type UpdateUserInput = {
  id: Scalars['ID']
  firstName?: Maybe<Scalars['String']>
  lastName?: Maybe<Scalars['String']>
  email?: Maybe<Scalars['String']>
}

export type User = {
  id: Scalars['ID']
  firstName: Scalars['String']
  lastName: Scalars['String']
  email: Scalars['String']
  notebooks?: Maybe<Array<Maybe<Notebook>>>
  createdAt: Scalars['String']
  updatedAt: Scalars['String']
}
export type ReadNoteQueryVariables = {
  id: Scalars['ID']
}

export type ReadNoteQuery = { __typename?: 'Query' } & {
  readNote: Maybe<{ __typename?: 'Note' } & NoteFragment>
}

export type ReadNotebookQueryVariables = {
  id: Scalars['ID']
}

export type ReadNotebookQuery = { __typename?: 'Query' } & {
  readNotebook: Maybe<
    { __typename?: 'Notebook' } & {
      notes: Maybe<Array<Maybe<{ __typename?: 'Note' } & NoteFragment>>>
    } & NotebookFragment
  >
}

export type ListNotebooksQueryVariables = {
  filter?: Maybe<ModelNotebookFilterInput>
  limit?: Maybe<Scalars['Int']>
  offset?: Maybe<Scalars['Int']>
}

export type ListNotebooksQuery = { __typename?: 'Query' } & {
  listNotebooks: Maybe<
    { __typename?: 'ModelNotebookConnection' } & {
      items: Maybe<Array<Maybe<{ __typename?: 'Notebook' } & NotebookFragment>>>
    }
  >
}

export type NoteFragment = { __typename?: 'Note' } & Pick<
  Note,
  'id' | 'title' | 'markdown' | 'excerpt' | 'createdAt' | 'updatedAt'
>

export type NotebookFragment = { __typename?: 'Notebook' } & Pick<
  Notebook,
  'id' | 'title' | 'createdAt' | 'updatedAt'
>

export type UserFragment = { __typename?: 'User' } & Pick<
  User,
  'id' | 'firstName' | 'lastName' | 'email' | 'createdAt' | 'updatedAt'
>

import gql from 'graphql-tag'
import * as React from 'react'
import * as ReactApollo from 'react-apollo'
export const noteFragmentDoc = gql`
  fragment note on Note {
    id
    title
    markdown
    excerpt
    createdAt
    updatedAt
  }
`
export const notebookFragmentDoc = gql`
  fragment notebook on Notebook {
    id
    title
    createdAt
    updatedAt
  }
`
export const userFragmentDoc = gql`
  fragment user on User {
    id
    firstName
    lastName
    email
    createdAt
    updatedAt
  }
`
export const ReadNoteDocument = gql`
  query ReadNote($id: ID!) {
    readNote(id: $id) {
      ...note
    }
  }
  ${noteFragmentDoc}
`

export class ReadNoteComponent extends React.Component<
  Partial<ReactApollo.QueryProps<ReadNoteQuery, ReadNoteQueryVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<ReadNoteQuery, ReadNoteQueryVariables>
        query={ReadNoteDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type ReadNoteProps<TChildProps = {}> = Partial<
  ReactApollo.DataProps<ReadNoteQuery, ReadNoteQueryVariables>
> &
  TChildProps
export function withReadNote<TProps, TChildProps = {}>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        ReadNoteQuery,
        ReadNoteQueryVariables,
        ReadNoteProps<TChildProps>
      >
    | undefined
) {
  return ReactApollo.withQuery<
    TProps,
    ReadNoteQuery,
    ReadNoteQueryVariables,
    ReadNoteProps<TChildProps>
  >(ReadNoteDocument, operationOptions)
}
export const ReadNotebookDocument = gql`
  query ReadNotebook($id: ID!) {
    readNotebook(id: $id) {
      ...notebook
      notes {
        ...note
      }
    }
  }
  ${notebookFragmentDoc}
  ${noteFragmentDoc}
`

export class ReadNotebookComponent extends React.Component<
  Partial<ReactApollo.QueryProps<ReadNotebookQuery, ReadNotebookQueryVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<ReadNotebookQuery, ReadNotebookQueryVariables>
        query={ReadNotebookDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type ReadNotebookProps<TChildProps = {}> = Partial<
  ReactApollo.DataProps<ReadNotebookQuery, ReadNotebookQueryVariables>
> &
  TChildProps
export function withReadNotebook<TProps, TChildProps = {}>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        ReadNotebookQuery,
        ReadNotebookQueryVariables,
        ReadNotebookProps<TChildProps>
      >
    | undefined
) {
  return ReactApollo.withQuery<
    TProps,
    ReadNotebookQuery,
    ReadNotebookQueryVariables,
    ReadNotebookProps<TChildProps>
  >(ReadNotebookDocument, operationOptions)
}
export const ListNotebooksDocument = gql`
  query ListNotebooks(
    $filter: ModelNotebookFilterInput
    $limit: Int
    $offset: Int
  ) {
    listNotebooks(filter: $filter, limit: $limit, offset: $offset) {
      items {
        ...notebook
      }
    }
  }
  ${notebookFragmentDoc}
`

export class ListNotebooksComponent extends React.Component<
  Partial<
    ReactApollo.QueryProps<ListNotebooksQuery, ListNotebooksQueryVariables>
  >
> {
  render() {
    return (
      <ReactApollo.Query<ListNotebooksQuery, ListNotebooksQueryVariables>
        query={ListNotebooksDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type ListNotebooksProps<TChildProps = {}> = Partial<
  ReactApollo.DataProps<ListNotebooksQuery, ListNotebooksQueryVariables>
> &
  TChildProps
export function withListNotebooks<TProps, TChildProps = {}>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        ListNotebooksQuery,
        ListNotebooksQueryVariables,
        ListNotebooksProps<TChildProps>
      >
    | undefined
) {
  return ReactApollo.withQuery<
    TProps,
    ListNotebooksQuery,
    ListNotebooksQueryVariables,
    ListNotebooksProps<TChildProps>
  >(ListNotebooksDocument, operationOptions)
}
export interface IntrospectionResultData {
  __schema: {
    types: {
      kind: string
      name: string
      possibleTypes: {
        name: string
      }[]
    }[]
  }
}

const result: IntrospectionResultData = {
  __schema: {
    types: [],
  },
}

export default result
