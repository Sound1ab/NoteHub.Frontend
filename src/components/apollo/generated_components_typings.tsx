/* eslint-disable */
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** The `Upload` scalar type represents a file upload. */
  Upload: any;
};


export type Query = {
  __typename?: 'Query';
  activeRetextSettings?: Maybe<Array<Retext_Settings>>;
  isDarkMode: Scalars['Boolean'];
  listRepos?: Maybe<Array<Repo>>;
  login: Scalars['String'];
  logout: Scalars['String'];
  readConfiguration?: Maybe<Configuration>;
  readFile?: Maybe<File>;
  readFiles?: Maybe<Array<File>>;
  readGithubUser?: Maybe<GithubUser>;
  readGithubUserAccessToken: Scalars['String'];
  readImage?: Maybe<File>;
  readRepo?: Maybe<Repo>;
  refresh?: Maybe<Scalars['String']>;
};


export type QueryReadFileArgs = {
  path: Scalars['String'];
};


export type QueryReadGithubUserAccessTokenArgs = {
  code: Scalars['String'];
  state: Scalars['String'];
};


export type QueryReadImageArgs = {
  path: Scalars['String'];
};


export type QueryReadRepoArgs = {
  name: Scalars['String'];
};

export type File = {
  __typename?: 'File';
  id: Scalars['ID'];
  path: Scalars['String'];
  type: Node_Type;
  sha: Scalars['String'];
  url: Scalars['String'];
  filename?: Maybe<Scalars['String']>;
  content?: Maybe<Scalars['String']>;
};

export enum Node_Type {
  File = 'FILE',
  Folder = 'FOLDER',
  User = 'USER'
}

export type Repo = {
  __typename?: 'Repo';
  id: Scalars['Int'];
  node_id: Scalars['String'];
  name: Scalars['String'];
  full_name: Scalars['String'];
  description?: Maybe<Scalars['String']>;
  private: Scalars['Boolean'];
  updated_at: Scalars['String'];
};

export type GithubUser = {
  __typename?: 'GithubUser';
  id: Scalars['Int'];
  login: Scalars['String'];
  avatar_url: Scalars['String'];
  html_url: Scalars['String'];
  name?: Maybe<Scalars['String']>;
  configuration?: Maybe<Configuration>;
};

export type Configuration = {
  __typename?: 'Configuration';
  id: Scalars['String'];
  connectedRepos?: Maybe<Array<Scalars['String']>>;
};

export type Mutation = {
  __typename?: 'Mutation';
  createFile?: Maybe<File>;
  updateFile?: Maybe<File>;
  deleteFile?: Maybe<File>;
  moveFile?: Maybe<File>;
  createImage?: Maybe<File>;
  updateImage?: Maybe<File>;
  deleteImage?: Maybe<File>;
  createSignedUrl?: Maybe<Scalars['String']>;
  createRepo?: Maybe<Repo>;
  updateRepo?: Maybe<Repo>;
  deleteRepo?: Maybe<Repo>;
  updateConfiguration?: Maybe<Configuration>;
};


export type MutationCreateFileArgs = {
  input: CreateFileInput;
};


export type MutationUpdateFileArgs = {
  input: UpdateFileInput;
};


export type MutationDeleteFileArgs = {
  input: DeleteFileInput;
};


export type MutationMoveFileArgs = {
  input: MoveFileInput;
};


export type MutationCreateImageArgs = {
  input: CreateFileInput;
};


export type MutationUpdateImageArgs = {
  input: UpdateFileInput;
};


export type MutationDeleteImageArgs = {
  input: DeleteFileInput;
};


export type MutationUpdateRepoArgs = {
  input: UpdateRepoInput;
};


export type MutationDeleteRepoArgs = {
  input: DeleteRepoInput;
};


export type MutationUpdateConfigurationArgs = {
  input: UpdateConfigurationInput;
};

export type CreateFileInput = {
  path: Scalars['String'];
  content?: Maybe<Scalars['String']>;
  retextSettings?: Maybe<Array<Retext_Settings>>;
};

export enum Retext_Settings {
  Spell = 'SPELL',
  Equality = 'EQUALITY',
  IndefiniteArticle = 'INDEFINITE_ARTICLE',
  RepeatedWords = 'REPEATED_WORDS',
  Readability = 'READABILITY'
}

export type UpdateFileInput = {
  path: Scalars['String'];
  content?: Maybe<Scalars['String']>;
  retextSettings?: Maybe<Array<Retext_Settings>>;
};

export type DeleteFileInput = {
  path: Scalars['String'];
};

export type MoveFileInput = {
  path: Scalars['String'];
  newPath: Scalars['String'];
};

export type UpdateRepoInput = {
  name: Scalars['String'];
  description?: Maybe<Scalars['String']>;
  private?: Maybe<Scalars['Boolean']>;
};

export type DeleteRepoInput = {
  name: Scalars['String'];
};

export type UpdateConfigurationInput = {
  connectedRepos?: Maybe<Array<Scalars['String']>>;
};

export type Links = {
  __typename?: 'Links';
  html: Scalars['String'];
};

export type ModelFileConnection = {
  __typename?: 'ModelFileConnection';
  items: Array<File>;
};

export type ModelRepoConnection = {
  __typename?: 'ModelRepoConnection';
  items: Array<Repo>;
};

export enum CacheControlScope {
  Public = 'PUBLIC',
  Private = 'PRIVATE'
}


export type TreeFileFragment = (
  { __typename?: 'File' }
  & Pick<File, 'id' | 'path' | 'type' | 'url' | 'sha'>
);

export type FileFragment = (
  { __typename?: 'File' }
  & Pick<File, 'filename' | 'content'>
  & TreeFileFragment
);

export type RepoFragment = (
  { __typename?: 'Repo' }
  & Pick<Repo, 'name' | 'description' | 'private' | 'full_name' | 'updated_at'>
);

export type ConfigurationFragment = (
  { __typename?: 'Configuration' }
  & Pick<Configuration, 'id' | 'connectedRepos'>
);

export type GithubUserFragment = (
  { __typename?: 'GithubUser' }
  & Pick<GithubUser, 'id' | 'login' | 'avatar_url' | 'html_url' | 'name'>
  & { configuration?: Maybe<(
    { __typename?: 'Configuration' }
    & ConfigurationFragment
  )> }
);

export type LoginQueryVariables = Exact<{ [key: string]: never; }>;


export type LoginQuery = (
  { __typename?: 'Query' }
  & Pick<Query, 'login'>
);

export type LogoutQueryVariables = Exact<{ [key: string]: never; }>;


export type LogoutQuery = (
  { __typename?: 'Query' }
  & Pick<Query, 'logout'>
);

export type ReadConfigurationQueryVariables = Exact<{ [key: string]: never; }>;


export type ReadConfigurationQuery = (
  { __typename?: 'Query' }
  & { readConfiguration?: Maybe<(
    { __typename?: 'Configuration' }
    & ConfigurationFragment
  )> }
);

export type UpdateConfigurationMutationVariables = Exact<{
  input: UpdateConfigurationInput;
}>;


export type UpdateConfigurationMutation = (
  { __typename?: 'Mutation' }
  & { updateConfiguration?: Maybe<(
    { __typename?: 'Configuration' }
    & ConfigurationFragment
  )> }
);

export type CreateSignedUrlMutationVariables = Exact<{ [key: string]: never; }>;


export type CreateSignedUrlMutation = (
  { __typename?: 'Mutation' }
  & Pick<Mutation, 'createSignedUrl'>
);

export type CreateRepoMutationVariables = Exact<{ [key: string]: never; }>;


export type CreateRepoMutation = (
  { __typename?: 'Mutation' }
  & { createRepo?: Maybe<(
    { __typename?: 'Repo' }
    & RepoFragment
  )> }
);

export type ListReposQueryVariables = Exact<{ [key: string]: never; }>;


export type ListReposQuery = (
  { __typename?: 'Query' }
  & { listRepos?: Maybe<Array<(
    { __typename?: 'Repo' }
    & Pick<Repo, 'id' | 'name' | 'full_name'>
  )>> }
);

export type ReadRepoQueryVariables = Exact<{
  name: Scalars['String'];
}>;


export type ReadRepoQuery = (
  { __typename?: 'Query' }
  & { readRepo?: Maybe<(
    { __typename?: 'Repo' }
    & RepoFragment
  )> }
);

export type ReadRepoLazyQueryVariables = Exact<{
  name: Scalars['String'];
}>;


export type ReadRepoLazyQuery = (
  { __typename?: 'Query' }
  & { readRepo?: Maybe<(
    { __typename?: 'Repo' }
    & RepoFragment
  )> }
);

export type ReadGithubUserQueryVariables = Exact<{ [key: string]: never; }>;


export type ReadGithubUserQuery = (
  { __typename?: 'Query' }
  & { readGithubUser?: Maybe<(
    { __typename?: 'GithubUser' }
    & GithubUserFragment
  )> }
);

export type ReadGithubUserAccessTokenQueryVariables = Exact<{
  code: Scalars['String'];
  state: Scalars['String'];
}>;


export type ReadGithubUserAccessTokenQuery = (
  { __typename?: 'Query' }
  & Pick<Query, 'readGithubUserAccessToken'>
);

export type Unnamed_1_QueryVariables = Exact<{ [key: string]: never; }>;


export type Unnamed_1_Query = (
  { __typename?: 'Query' }
  & Pick<Query, 'refresh'>
);


      export interface IntrospectionResultData {
        __schema: {
          types: {
            kind: string;
            name: string;
            possibleTypes: {
              name: string;
            }[];
          }[];
        };
      }
      const result: IntrospectionResultData = {
  "__schema": {
    "types": []
  }
};
      export default result;
    