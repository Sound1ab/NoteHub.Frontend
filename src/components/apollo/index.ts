export * from './generated_components_typings'
