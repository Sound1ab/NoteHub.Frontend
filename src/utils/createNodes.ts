import {
  GitNode,
  Node_Type,
} from '../components/apollo/generated_components_typings'
import { ITreeNode } from '../types'
import { isFile } from './isFile'

export function createFolderNode(
  name: string,
  path: string,
  toggled: boolean,
  isOptimistic: boolean
) {
  return {
    children: [],
    name,
    path,
    toggled,
    type: Node_Type.Folder,
    isOptimistic,
  }
}

export function createFileNode(
  name: string,
  path: string,
  isOptimistic: boolean
) {
  return {
    name,
    path,
    toggled: false,
    type: Node_Type.File,
    isOptimistic,
  }
}

function getNode(nodes: ITreeNode[], slug: string) {
  return nodes.find((node) => node.name === slug)
}

function createNode(
  type: Node_Type,
  slug: string,
  path: string,
  listOfToggledPaths: Set<string>,
  isOptimistic = false
) {
  const isFile = type === Node_Type.File

  return isFile
    ? createFileNode(slug, path, isOptimistic)
    : createFolderNode(slug, path, listOfToggledPaths.has(path), isOptimistic)
}

interface IInsertNodes {
  path: string[]
  parentNode: ITreeNode
  gitNode: GitNode
  currentPath?: string
  listOfToggledPaths: Set<string>
}

export function insertNodeIntoParentNode({
  path,
  parentNode,
  gitNode,
  currentPath = '',
  listOfToggledPaths,
}: IInsertNodes): void {
  if (path.length === 1) {
    const [slug] = path

    const children = parentNode?.children ? parentNode.children : []

    // Setting the sha value to optimistic during moveFile update function
    // optimistic update. This allows us to determine if the change is in flight
    // in NodeItem
    const node = createNode(
      gitNode.type,
      slug,
      gitNode.path,
      listOfToggledPaths,
      gitNode.sha === 'optimistic'
    )

    parentNode.children = [...children, node]

    return
  }

  if (!parentNode.children) {
    throw new Error('CurrentNode has no children')
  }

  const [slug, ...rest] = path

  const nextNode = getNode(parentNode.children, slug)

  const nextPath = currentPath ? `${currentPath}/${slug}` : slug

  // If no next node the optimistic update has run and generated a file without
  // a parent gitNode. Create parent node and retry.
  if (!nextNode) {
    const parentPath = parentNode.path
      ? `${parentNode.path}/${nextPath}`
      : nextPath

    parentNode.children = [
      ...parentNode.children,
      createFolderNode(
        slug,
        parentPath,
        listOfToggledPaths.has(parentPath),
        false // Child is optimistic not this node
      ),
    ]

    return insertNodeIntoParentNode({
      path,
      parentNode,
      gitNode,
      currentPath,
      listOfToggledPaths,
    })
  }

  return insertNodeIntoParentNode({
    path: rest,
    parentNode: nextNode,
    gitNode,
    currentPath: nextPath,
    listOfToggledPaths,
  })
}

export function createNodes(
  gitNodes: GitNode[],
  listOfToggledPaths: Set<string>
) {
  return gitNodes.reduce<ITreeNode[]>((acc, gitNode) => {
    const { path } = gitNode

    const [rootSlug, ...restOfPath] = path.split('/')

    const parentNode = getNode(acc, rootSlug)

    if (parentNode) {
      insertNodeIntoParentNode({
        path: restOfPath,
        parentNode,
        gitNode,
        listOfToggledPaths,
      })

      return acc
    }

    if (isFile(path) && restOfPath.length > 0) {
      // If no parentNode and its a file inside a folder, the optimistic update has
      // run and generated a file without a parent gitNode. Create a parent node
      // and insert file node. Any other missing parent nodes further along the path
      // will be created inside insertNodeIntoParentNode.
      const newParentNode = createNode(
        Node_Type.Folder,
        rootSlug,
        rootSlug,
        listOfToggledPaths
      )

      insertNodeIntoParentNode({
        path: restOfPath,
        parentNode: newParentNode,
        gitNode,
        listOfToggledPaths,
      })

      return [...acc, newParentNode]
    } else {
      // Else we just have a top level file/folder that we need to create.
      return [
        ...acc,
        createNode(gitNode.type, rootSlug, path, listOfToggledPaths),
      ]
    }
  }, [])
}
