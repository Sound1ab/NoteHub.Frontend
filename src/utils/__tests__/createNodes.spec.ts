import { Node_Type } from '../../components/apollo/generated_components_typings'
import { createNodes } from '../createNodes'

export const rootFile = {
  path: 'README.md',
  type: Node_Type.File,
  isOptimistic: false,
}

export const rootFolder = {
  path: 'folder',
  type: Node_Type.Folder,
  isOptimistic: false,
}

const fileDepthOne = {
  path: 'folder/file.md',
  type: Node_Type.File,
  isOptimistic: false,
}

const folderDepthOne = {
  path: 'folder/folder2',
  type: Node_Type.Folder,
  isOptimistic: false,
}

const fileDepthTwo = {
  path: 'folder/folder2/file2.md',
  type: Node_Type.File,
  isOptimistic: false,
}

describe('createNodes', () => {
  it('should create a flattened object of nodes with children', async () => {
    const githubTreeData = [
      rootFile,
      rootFolder,
      fileDepthOne,
      folderDepthOne,
      fileDepthTwo,
    ]

    const treeObject = createNodes(githubTreeData, new Set())

    expect(treeObject).toMatchSnapshot()
  })

  it('should construct missing parent nodes that may have resulted from an optimistic update', async () => {
    const githubTreeData = [rootFile, fileDepthOne, fileDepthTwo]

    const treeObject = createNodes(githubTreeData, new Set())

    expect(treeObject).toMatchSnapshot()
  })

  it('should handle duplicate filenames at the top level', async () => {
    const githubTreeData = [rootFile, rootFile, fileDepthOne, fileDepthTwo]

    const treeObject = createNodes(githubTreeData, new Set())

    expect(treeObject).toMatchSnapshot()
  })
})
