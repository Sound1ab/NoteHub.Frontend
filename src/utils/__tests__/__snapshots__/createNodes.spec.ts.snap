// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNodes should construct missing parent nodes that may have resulted from an optimistic update 1`] = `
Array [
  Object {
    "id": "README.md",
    "isOptimistic": false,
    "name": "README.md",
    "path": "README.md",
    "type": "FILE",
  },
  Object {
    "children": Array [
      Object {
        "id": "folder/file.md",
        "isOptimistic": false,
        "name": "file.md",
        "path": "folder/file.md",
        "type": "FILE",
      },
      Object {
        "children": Array [
          Object {
            "id": "folder/folder2/file2.md",
            "isOptimistic": false,
            "name": "file2.md",
            "path": "folder/folder2/file2.md",
            "type": "FILE",
          },
        ],
        "id": "folder/folder2",
        "isOptimistic": false,
        "name": "folder2",
        "path": "folder/folder2",
        "toggled": false,
        "type": "FOLDER",
      },
    ],
    "id": "folder",
    "isOptimistic": false,
    "name": "folder",
    "path": "folder",
    "toggled": false,
    "type": "FOLDER",
  },
]
`;

exports[`createNodes should create a flattened object of nodes with children 1`] = `
Array [
  Object {
    "id": "README.md",
    "isOptimistic": false,
    "name": "README.md",
    "path": "README.md",
    "type": "FILE",
  },
  Object {
    "children": Array [
      Object {
        "id": "folder/file.md",
        "isOptimistic": false,
        "name": "file.md",
        "path": "folder/file.md",
        "type": "FILE",
      },
      Object {
        "children": Array [
          Object {
            "id": "folder/folder2/file2.md",
            "isOptimistic": false,
            "name": "file2.md",
            "path": "folder/folder2/file2.md",
            "type": "FILE",
          },
        ],
        "id": "folder/folder2",
        "isOptimistic": false,
        "name": "folder2",
        "path": "folder/folder2",
        "toggled": false,
        "type": "FOLDER",
      },
    ],
    "id": "folder",
    "isOptimistic": false,
    "name": "folder",
    "path": "folder",
    "toggled": false,
    "type": "FOLDER",
  },
]
`;

exports[`createNodes should handle duplicate filenames at the top level 1`] = `
Array [
  Object {
    "id": "README.md",
    "isOptimistic": false,
    "name": "README.md",
    "path": "README.md",
    "type": "FILE",
  },
  Object {
    "children": Array [
      Object {
        "id": "folder/file.md",
        "isOptimistic": false,
        "name": "file.md",
        "path": "folder/file.md",
        "type": "FILE",
      },
      Object {
        "children": Array [
          Object {
            "id": "folder/folder2/file2.md",
            "isOptimistic": false,
            "name": "file2.md",
            "path": "folder/folder2/file2.md",
            "type": "FILE",
          },
        ],
        "id": "folder/folder2",
        "isOptimistic": false,
        "name": "folder2",
        "path": "folder/folder2",
        "toggled": false,
        "type": "FOLDER",
      },
    ],
    "id": "folder",
    "isOptimistic": false,
    "name": "folder",
    "path": "folder",
    "toggled": false,
    "type": "FOLDER",
  },
]
`;
